### PR TITLE
MGMT-12756 cluster name cannot use dot

### PR DESCRIPTION
**Pull request by celdrake**
https://issues.redhat.com/browse/MGMT-12756

Prevent cluster name to contain "dot" character
Similar to https://github.com/openshift-assisted/assisted-ui-lib/pull/1786/files

plus some Eslint fixes

### CODE REVIEW

**Comment by ammont82** [comment]
This validation affects CIM side,not? Maybe we need a review from @jtomasek

**Review by ammont82 — DISMISSED** [review]

**Comment by celdrake** [comment]
> This validation affects CIM side,not? Maybe we need a review from @jtomasek

It shouldn't, but you are right that it could. I have asked Jirka for review.

**Comment by mareklibra** [comment]
There was similar work done recently for CIM, I think we should unify: https://github.com/openshift-assisted/assisted-ui-lib/pull/1786

File: src/common/components/ui/formik/validationSchemas.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ const NAME_CHARS_REGEX = /^[a-z0-9-.]*$/;
 const CLUSTER_NAME_START_END_REGEX = /^[a-z0-9](.*[a-z0-9])?$/;
 // NOTE: based on https://github.com/openshift/assisted-service/blob/master/internal/cluster/validations/validations.go#L32
 const CLUSTER_NAME_REGEX =
-  /^[.-a-z0-9]?(([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*)*)[.-a-z0-9]?$/;
+  /^[-a-z0-9]?(([a-z0-9]([-a-z0-9]*[a-z0-9])?([a-z0-9]([-a-z0-9]*[a-z0-9])?)*)*)[-a-z0-9]?$/;
 const CLUSTER_NAME_NO_DOT_REGEX = /^[a-z0-9-]*$/;
```
Comment (mareklibra) [inline src/common/components/ui/formik/validationSchemas.ts:31]: Isn't `CLUSTER_NAME_REGEX` what should be used?

Comment (celdrake) [reply src/common/components/ui/formik/validationSchemas.ts:31]: The interactive checker in OCM has more restrictions than what CIM allows now.
I'm guessing the regular expression could be simplified further but it was hard to understand the parts.

checker

Comment (mareklibra) [reply src/common/components/ui/formik/validationSchemas.ts:31]: I don't think `-test` is either a valid name or beginning of an URL.

I think both CIM and OCM should be unified - none of them use `dots`. And `dash (-)` is allowed in both cases. Max length is equal.

In both cases we end up with a k8s resource named by that string and later on building an URL out of it.

Comment (celdrake) [reply src/common/components/ui/formik/validationSchemas.ts]: @mareklibra I think you are right, in fact the `nameValidationSchema` uses the validation that the first and last characters are letters for both CIM and OCM.

PTAL, I set the same Regexp for both, one which doesn't allow the `dash` to be either at the beginning or end of the name.

**Review by ammont82 — APPROVED** [review]

File: src/common/components/ui/formik/validationSchemas.ts [hunk 2]
```diff
@@ -26,9 +26,8 @@ const NAME_START_END_REGEX = /^[a-z0-9](.*[a-z0-9])?$/;
 const NAME_CHARS_REGEX = /^[a-z0-9-.]*$/;
 const CLUSTER_NAME_START_END_REGEX = /^[a-z0-9](.*[a-z0-9])?$/;
 // NOTE: based on https://github.com/openshift/assisted-service/blob/master/internal/cluster/validations/validations.go#L32
-const CLUSTER_NAME_REGEX =
-  /^[.-a-z0-9]?(([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*)*)[.-a-z0-9]?$/;
-const CLUSTER_NAME_NO_DOT_REGEX = /^[a-z0-9-]*$/;
+const CLUSTER_NAME_NO_DOT_REGEX =
+  /^[a-z0-9]?(([a-z0-9]([-a-z0-9]*[a-z0-9])?([a-z0-9]([-a-z0-9]*[a-z0-9])?)*)*)[a-z0-9]?$/;
```
Comment (mareklibra) [inline src/common/components/ui/formik/validationSchemas.ts:30]: When the dots are removed, this expression is way too complicated.

Without dots, we can check for kind of following one (not tested!!):
`/^[a-z0-9]?[a-z0-9-]*$/`

(Must start at an alphanumeric, followed by any count of alphanumerics or dashes. Can end at dash.)

Comment (celdrake) [reply src/common/components/ui/formik/validationSchemas.ts:30]: I agree it looks too complicated, let me see if I can refine it.
It should not be allowed to end in a dash, as there is a separate check for it (via `CLUSTER_NAME_START_END_REGEX`).
name

Comment (mareklibra) [reply src/common/components/ui/formik/validationSchemas.ts:30]: ok, then. I wasn't sure. Should I play with putting the expression together?

Comment (celdrake) [reply src/common/components/ui/formik/validationSchemas.ts]: @mareklibra This last comment made me think, we don't need the main RegExp to cover all cases, given that we're using `CLUSTER_NAME_START_END_REGEX` to identify the start and end of the name.

Therefore the "main" RegExp only cares about defining the allowed characters. I put back the simple RegExp expression that @jtomasek had inially written, only I renamed it for clarity.

See how it works now:
https://user-images.githubusercontent.com/829045/208850959-9dcd59f7-723b-4547-b122-bd1096fcb1f6.mp4

Comment (mareklibra) [reply src/common/components/ui/formik/validationSchemas.ts]: You are right, this is better when we already have it split.

**Review by jtomasek — APPROVED** [review]

**Review by mareklibra — APPROVED** [review]

**Pull request merged**